File: medplum-java/medplum-java-model/src/main/java/com/fhir/SubstanceReferenceInformation_Target.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSubstanceReferenceInformation_Target.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SubstanceReferenceInformation_Target {
  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<String> amountString();

  Optional<List<Reference>> source();

  Optional<Range> amountRange();

  Optional<CodeableConcept> organismType();

  Optional<Identifier> target();

  Optional<CodeableConcept> organism();

  Optional<CodeableConcept> type();

  Optional<Quantity> amountQuantity();

  Optional<CodeableConcept> interaction();

  Optional<CodeableConcept> amountType();

  Optional<String> id();

  static ImmutableSubstanceReferenceInformation_Target.Builder builder() {
    return ImmutableSubstanceReferenceInformation_Target.builder();
  }
}

File: medplum-typescript/src/model/fhir/SubstanceReferenceInformation_Target.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Identifier } from "./Identifier";
import { Quantity } from "./Quantity";
import { Range } from "./Range";
import { Reference } from "./Reference";

export interface SubstanceReferenceInformation_Target {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    target: Identifier | null | undefined;
    type: CodeableConcept | null | undefined;
    interaction: CodeableConcept | null | undefined;
    organism: CodeableConcept | null | undefined;
    organismType: CodeableConcept | null | undefined;
    amountQuantity: Quantity | null | undefined;
    amountRange: Range | null | undefined;
    amountString: string | null | undefined;
    amountType: CodeableConcept | null | undefined;
    source: Reference[] | null | undefined;
}
